app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt

```diff
@@ -283,6 +290,21 @@ class StateFragmentPresenter @Inject constructor(
     }
 
     val ephemeralState = result.getOrThrow()
+    val shouldSplit = splitScreenManager.shouldSplit(ephemeralState.state.interaction.id)
+
+    if (shouldSplit) {
+      viewModel.shouldSplitView.set(true)
+      binding.rhsStateRecyclerView.visibility = View.VISIBLE
+      val params = binding.centerGuideline.layoutParams as ConstraintLayout.LayoutParams
+      params.guidePercent = 0.5f
```

Comment:
Introduce a boolean ObservableField in viewModel to and update that value in viewModel, such that the value in xml directly gets updated and you can also control the visibility.

Reply:
Done